reports.php

```diff
@@ -229,12 +230,49 @@
 		$fileinfo = pathinfo(basename($report_path));
 		$images = array('jpeg', 'jpg', 'png', 'gif');
 		$text = array('txt', 'out', 'log');
+                $markdown = array('md');
+                $pdf = array('pdf');
+                $excel = array('xlsx', 'xls');
 		if(in_array($fileinfo['extension'], $images)){
 			echo '<p style="text-align:center;"><img src="ajax/send_file.php?path='.$report_path.'"></p>';
 		} else if(in_array($fileinfo['extension'], $text)){
 			echo '<pre>'.file_get_contents($data_root.$report_path).'</pre>';
+                } else if(in_array($fileinfo['extension'], $markdown)){
+                        include('ParseDown/Parsedown.php');
+                        $Parsedown = new Parsedown();
+                        echo '<span>'.$Parsedown->text(file_get_contents($data_root.$report_path)).'</span>';
+                } else if(in_array($fileinfo['extension'], $excel)){
+                        require_once dirname(__FILE__) . '/PHPExcel/Classes/PHPExcel/IOFactory.php';
+                        $inputFileType = PHPExcel_IOFactory::identify ( $data_root.$report_path );
+                        $readerObj = PHPExcel_IOFactory::createReader ( $inputFileType );
+                        $excelFile = $readerObj->load ( $data_root.$report_path );
+                        $excelFile->setActiveSheetIndex ( 0 );
+                        $activeSheet = $excelFile->getActiveSheet();
+                        $highestRow = $activeSheet->getHighestRow();
+                        $highestColumn = $activeSheet->getHighestColumn();
+                        $highestColumn = ord(strtolower($highestColumn)) - 96;
+                        $currentRow = 1;
+                        echo '<pre>'.$report_path.' Rows='.$highestRow." Columns=".$highestColumn.'<BR><TABLE BORDER=1>';
+                        while ( $currentRow < $highestRow ) {
+
+                                $currentColumn = 0;
+                                echo '<TR>';
+                                while ( $currentColumn < $highestColumn ) {
+
+                        		$value = $activeSheet->getCellByColumnAndRow ( $currentColumn, $currentRow );
+                                        $value = $value."";
+                                        echo "<TD>".$value."</TD>";
+                                        $currentColumn++;
+                                      }
+                                echo '</TR>';
+                                $currentRow++;
+                              }
+                        echo "</TABLE></pre>";
+                } else if(in_array($fileinfo['extension'], $pdf)){
+                        $url_short = $data_root_short;
+                        echo '<pre><embed src="http://ctr-bfx.pdn.private.cam.ac.uk/'.$url_short.$report_path.'" width="100%" height="1000px" type="application/pdf"></pre>'; 

```

Comment:
Found some more hardcoded stuff. Instead of the new variable `$data_root_short` here, can't you just use `$labrador_url` for the whole URL? (`http://server/labrador/`)